go/pkg/tree/tree.go

```diff
@@ -46,6 +46,16 @@ type Stats struct {
 	// TODO(olaola): number of FileMetadata cache hits/misses go here.
 }
 
+// ConstructTreeConfig controls the behavior when constructing a tree from the local filesystem.
+type ConstructTreeConfig struct {
+	// By default, symlinks are implicitly treated as a regular file, with the
+	// link metadata stripped away. Setting this to true causes the symlinks to
+	// be distinguished from the normal files.
+	DistinguishSymlinkFromFile bool
+	// If true, the dangling symlinks will not be treated as an error.
+	AllowDanglingSymlink bool
```

Comment:
I prefer to remove this and allow dangling symlink when symlink is enabled.

Reply:
I don't think so, because the default client behavior is to treat this as an error, and we want backward compatibility?

Reply:
Enabling symlink is new behavior, so I think there is no previous behavior we want to keep compatibility?

Reply:
I think these two are independent (i.e. it's fine if one enables `AllowDanglingSymlink` but not `EnableSymlink`). There is a check to return error for dangling symlinks:

https://github.com/bazelbuild/remote-apis-sdks/blob/af8c2eea6772090ab71c0eda356df2f011893626/go/pkg/tree/tree.go#L68-L70

Reply:
I see, but I still think it is better to make it small where user can configure. Adding `AllowDanglingSymlink` later is easy if we found it is necessary, but removing is hard.

Reply:
My understanding is that the SDK does not want to allow dangling symlinks by default, therefore this config. Let's see @ola-rozenfeld 's opinion?

Reply:
I've switched to the functional pattern. Meanwhile I also removed the config for the dangling symlink

Reply:
From https://github.com/bazelbuild/remote-apis-sdks/pull/188#issuecomment-702233019 I think the SDK team wants the handling of symlink and allowing dangling symlinks to be separate.

Reply:
Yes, I'd prefer to have these as separate options, to preserve backward compatibility. Also, make sure that the default options are clear and initialized correctly (AllowDanglingSymlink=true).
Also, I'd actually prefer the introduction of the options and the implementation to be in the same PR, it's easier to review that way IMO.

Reply:
Suggestion: now that tree is a part of Client after a recent PR, it would make a lot of sense to me to just make both of these Client Opts instead.

Reply:
Ack.